app/repositories/map_requests.py

```diff
@@ -0,0 +1,184 @@
+             VALUES (:map_id, :player_id, NOW(), :active)
+    """
+    params = {
+        "map_id": map_id,
+        "player_id": player_id,
+        "active": active,
+    }
+    rec_id = await app.state.services.database.execute(query, params)
+
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM map_requests
+         WHERE id = :id
+    """
+    params = {
+        "id": rec_id,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    assert rec is not None
+    return dict(rec)
+
+
+async def fetch_one(
+    id: Optional[int] = None,
+    map_id: Optional[int] = None,
+    player_id: Optional[int] = None,
+    active: Optional[int] = None,
+) -> Optional[dict[str, Any]]:
+    """Fetch a map request entry from the database."""
+    if id is None and map_id is None and player_id is None and active is None:
+        raise ValueError("Must provide at least one parameter.")
+
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM map_requests
+         WHERE id = COALESCE(:id, id)
+           AND map_id = COALESCE(:map_id, map_id)
+           AND player_id = COALESCE(:player_id, player_id)
+           AND active = COALESCE(:active, active)
+    """
+    params = {
+        "id": id,
+        "map_id": map_id,
+        "player_id": player_id,
+        "active": active,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    return dict(rec) if rec is not None else None
+
+
+async def fetch_count(
+    map_id: Optional[int] = None,
+    player_id: Optional[int] = None,
+    active: Optional[int] = None,
+) -> int:
+    """Fetch the number of map requests in the database."""
+    query = """\
+        SELECT COUNT(*) AS count
+          FROM map_requests
+        WHERE map_id = COALESCE(:map_id, map_id)
+          AND player_id = COALESCE(:player_id, player_id)
+          AND active = COALESCE(:active, active)
+    """
+    params = {
+        "map_id": map_id,
+        "player_id": player_id,
+        "active": active,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    assert rec is not None
+    return rec["count"]
+
+
+async def fetch_all(
+    map_id: Optional[int] = None,
+    player_id: Optional[int] = None,
+    active: Optional[int] = None,
+    page: Optional[int] = None,
+    page_size: Optional[int] = None,
+) -> list[dict[str, Any]]:
+    """Fetch a list of map requests from the database."""
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM map_requests
+         WHERE map_id = COALESCE(:map_id, map_id)
+          AND player_id = COALESCE(:player_id, player_id)
+          AND active = COALESCE(:active, active)
+    """
+    params = {
+        "map_id": map_id,
+        "player_id": player_id,
+        "active": active,
+    }
+    recs = await app.state.services.database.fetch_all(query, params)
+    return [dict(rec) for rec in recs]
+
+
+async def update(
+    map_ids: Optional[list[any]] = None,
```

Comment:
use  this and `from typing import Any`. `any` in python is a builtin function to check whether any elements in an iterable return true, e.g. `any([True, False])`
```suggestion
    map_ids: Optional[list[Any]] = None,
```